src/components/calcite-radio-button/calcite-radio-button.e2e.ts

```diff
@@ -245,6 +245,78 @@ describe("calcite-radio-button", () => {
     expect(changeEvent).toHaveReceivedEventTimes(1);
   });
 
+  it("appropriately triggers the custom internal focus and blur events on click", async () => {
+    const page = await newE2EPage();
+    await page.setContent(
+      `<calcite-radio-button></calcite-radio-button><calcite-radio-button id="two"></calcite-radio-button>`
+    );
+
+    const radio = await page.find("calcite-radio-button");
+    const radio2 = await page.find("calcite-radio-button#two");
+
+    const focusEvent = await radio.spyOnEvent("calciteInternalRadioButtonFocus");
+    const blurEvent = await radio.spyOnEvent("calciteInternalRadioButtonBlur");
+
+    expect(focusEvent).toHaveReceivedEventTimes(0);
+    expect(blurEvent).toHaveReceivedEventTimes(0);
+
+    await radio.click();
+
+    expect(focusEvent).toHaveReceivedEventTimes(1);
+    expect(blurEvent).toHaveReceivedEventTimes(0);
+
+    await radio2.click();
+
+    expect(focusEvent).toHaveReceivedEventTimes(1);
+    expect(blurEvent).toHaveReceivedEventTimes(1);
+  });
+
+  it("appropriately triggers the custom internal focus and blur events with keyboard", async () => {
```

Comment:
Nitpick: maybe this can be merged into the previous test as "appropriately triggers the custom internal focus and blur events". I'll leave this up to you.